### PR TITLE
Stack traces in jest parsing enabled

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
Other minor fixes/refactors across parsers

### CODE REVIEW

File: Agent.Plugins.UnitTests/JestTestResultParserTests/Resources/StackTraceTests/TestCase005.txt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      36 |     afterEach(() => {
+
+      at Object.toBe (unit_tests/sample.test.js:33:22)
+
+ FAIL  unit_tests/sample2.test.js
+  first set
+    √ Test method 3
+    × Test method 4 (1ms)
+    × Test method 5 (1ms)
+    ○ skipped 2 tests
```
Comment (singhsarab) [inline Agent.Plugins.UnitTests/JestTestResultParserTests/Resources/StackTraceTests/TestCase005.txt]: >skipped 2 tests [](start = 6, length = 15)

Are we missing this data?

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jest/States/JestParserStateExpectingTestResults.cs [hunk 2]
```diff
@@ -67,6 +70,12 @@ private Enum StackTraceStartMatched(Match match, AbstractParserStateContext stat
 
             var testResult = PrepareTestResult(TestOutcome.Failed, match);
             jestStateContext.TestRun.FailedTests.Add(testResult);
+            jestStateContext.CurrentStackTraceIndex = jestStateContext.TestRun.FailedTests.Count - 1;
+
+            // Expect the stack trace to not be more than 50 lines long
```
Comment (singhsarab) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jest/States/JestParserStateExpectingTestResults.cs]: Expected.

File: Agent.Plugins.UnitTests/Agent.Plugins.UnitTests.csproj [hunk 3]
```diff
@@ -29,6 +29,13 @@
     </Content>
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (singhsarab) [inline Agent.Plugins.UnitTests/Agent.Plugins.UnitTests.csproj]: Remove

Comment (ShreyasRmsft) [reply Agent.Plugins.UnitTests/Agent.Plugins.UnitTests.csproj]: done

File: Agent.Plugins.UnitTests/JestTestResultParserTests/Resources/StackTraceTests/TestCase005.txt (hunk 1 above)
Comment (ShreyasRmsft) [reply Agent.Plugins.UnitTests/JestTestResultParserTests/Resources/StackTraceTests/TestCase005.txt]: lemme ask nitin if the api supports uploading only skipped tests count

Comment (ShreyasRmsft) [reply Agent.Plugins.UnitTests/JestTestResultParserTests/Resources/StackTraceTests/TestCase005.txt]: this is support and we are anyhow picking this up from the summary. This is of no use to us

**Pull request merged**